src/metabase/search/filter.clj

```diff
@@ -155,4 +196,7 @@
 
       ;; build optional filters
       (int? created-by)
-      (#(build-optional-filter-query :created-by model % created-by)))))
+      (#(build-optional-filter-query :created-by model % created-by))
```

Comment:
It might be tidier to refactor this multimethod so that `build-optional-filter-query` has the query as it's first argument, and instead dispatches on its second and third arguments. I think that feels more natural in use, since for functions that modify maps the first argument is usually the map being modified e.g. `update` and `assoc`

Reply:
I thought about that but it makes the `defmethod` feel unnatural to have the first argument not part of the dispatch.

calling `(build-optional-filter-query :created-by model ...)` is easy to know what it dispatches on IMO.

Reply:
Okay, makes sense. Both ways feel unusual to me so I don't have a strong opinion on what to prefer